### PR TITLE
[WIP] Light: Add support for network destination driver

### PR DESCRIPTION
**Pull request by mitzkia**
This PR introduces supporting network destination driver.
We have decided to use netcat as a message collector on syslog-ng's output side.
Reasons behind using netcat: 
- almost a standard default package on every Linux systems
- netcat is already installed on our DBLD's devshell image where Light will run in almost all cases

Adding a default parametrized testcase for testing template() option in various cases.

### CODE REVIEW

**Comment by kira-syslogng** [comment]
Build SUCCESS

File: tests/python_functional/functional_tests/destination_drivers/network_destination/test_network_destination_transport.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    generator_source = config.create_example_msg_generator_source(num=counter, freq=0.0001, template=config.stringify(message))
+    if transport is not None:
+        network_destination = config.create_network_destination(ip="localhost", port=port_allocator(), transport=transport)
+    else:
+        network_destination = config.create_network_destination(ip="localhost", port=port_allocator())
+    config.create_logpath(statements=[generator_source, network_destination])
+
+    network_destination.start_listener()
+    syslog_ng.start(config)
+    assert network_destination.read_until_logs([message] * counter)
```
[the remark this answers is not in the document]
Comment (alltilla) [reply tests/python_functional/functional_tests/destination_drivers/network_destination/test_network_destination_transport.py:47]: https://github.com/syslog-ng/syslog-ng/pull/3668/files#diff-29a639ad3c4ced688d82d29e949c58666af18284a62dd60a953b352be721c864R44 should take care of it. Is it not working?

[the remark this answers is not in the document]
Comment (alltilla) [reply tests/python_functional/functional_tests/destination_drivers/network_destination/test_network_destination_transport.py:47]: Maybe that does not run when a testcase is finished :(

[the remark this answers is not in the document]
Comment (alltilla) [reply tests/python_functional/functional_tests/destination_drivers/network_destination/test_network_destination_transport.py:47]: Yes, I checked, that is the case.

We wanted to automate this, and first we have put it in the `NetworkIO`'s `__del__`, but unfortunately that did not work, as for an assertion failed case, python partially freed some resources at that point, which made our process's `terminate()` call fail.

The listener must stop after each testcase, I agree. Maybe the native implementation would work in `__del__`. I would be sad, if we had to manually set a cleanup for testcases like this, where it looks like that python could take care of it automatically. I hope, that we can find a solution for this problem in general.

[the remark this answers is not in the document]
Comment (MrAnno) [reply tests/python_functional/functional_tests/destination_drivers/network_destination/test_network_destination_transport.py:47]: pytest finalizers might help here.

**Comment by mitzkia** [comment]
@kira-syslogng test this please;

**Comment by kira-syslogng** [comment]
Build FAILURE

**Comment by MrAnno** [comment]
We're preparing a new branch.

**Pull request closed without merging**